### ISSUE TITLE
Signal 11 received and server down when inferring bert model

### ISSUE THREAD

**Issue by XZHUNZHUN** [triton-inference-server/server#4008]
### **Description**
I am testing out our BERT model but met some issues when sending requests to tritonserver. The model can be successfully loaded. And I tried using warmup but still not useful.
```I0303 02:20:51.218350 1 http_server.cc:2727] HTTP request: 0 /v2/models/bert
        I0303 02:20:51.218387 1 model_repository_manager.cc:615] GetInferenceBackend() 'bert' version -1
        I0303 02:20:51.218394 1 model_repository_manager.cc:571] VersionStates() 'bert'
        I0303 02:20:51.218740 1 http_server.cc:2727] HTTP request: 0 /v2/models/bert/config
        I0303 02:20:51.218760 1 model_repository_manager.cc:615] GetInferenceBackend() 'bert' version -1
        I0303 02:20:51.220923 1 http_server.cc:2727] HTTP request: 2 /v2/models/bert/infer
        I0303 02:20:51.220941 1 model_repository_manager.cc:615] GetInferenceBackend() 'bert' version -1
        I0303 02:20:51.220948 1 model_repository_manager.cc:615] GetInferenceBackend() 'bert' version -1
        I0303 02:20:51.220991 1 infer_request.cc:524] prepared: [0x0x7f975800b4d0] request id: 1, model: bert, requested version: -1, actual version: 1, flags: 0x0, correlation id: 0, batch size: 1, priority: 0, timeout (us): 0
        original inputs:
        [0x0x7f9758012aa8] input: segment_ids, type: INT32, original shape: [1,128], batch + shape: [1,128], shape: [128]
        [0x0x7f9758008638] input: label_ids, type: INT32, original shape: [1,128], batch + shape: [1,128], shape: [128]
        [0x0x7f97580128c8] input: input_mask, type: INT32, original shape: [1,128], batch + shape: [1,128], shape: [128]
        [0x0x7f9758012c38] input: input_ids, type: INT32, original shape: [1,128], batch + shape: [1,128], shape: [128]
        override inputs:
        inputs:
        [0x0x7f9758012c38] input: input_ids, type: INT32, original shape: [1,128], batch + shape: [1,128], shape: [128]
        [0x0x7f97580128c8] input: input_mask, type: INT32, original shape: [1,128], batch + shape: [1,128], shape: [128]
        [0x0x7f9758008638] input: label_ids, type: INT32, original shape: [1,128], batch + shape: [1,128], shape: [128]
        [0x0x7f9758012aa8] input: segment_ids, type: INT32, original shape: [1,128], batch + shape: [1,128], shape: [128]
        original requested outputs:
        cls_logits
        cls_loss
        cls_per_example_loss
        cls_probabilities
        requested outputs:
        cls_logits
        cls_loss
        cls_per_example_loss
        cls_probabilities
        
        I0303 02:20:51.221100 1 tensorflow.cc:2390] model bert, instance bert, executing 1 requests
        I0303 02:20:51.221130 1 tensorflow.cc:1564] TRITONBACKEND_ModelExecute: Running bert with 1 requests
        I0303 02:20:51.221196 1 tensorflow.cc:1816] TRITONBACKEND_ModelExecute: input 'input_ids' is GPU tensor: false
        I0303 02:20:51.221212 1 tensorflow.cc:1816] TRITONBACKEND_ModelExecute: input 'input_mask' is GPU tensor: false
        I0303 02:20:51.221230 1 tensorflow.cc:1816] TRITONBACKEND_ModelExecute: input 'label_ids' is GPU tensor: false
        I0303 02:20:51.221247 1 tensorflow.cc:1816] TRITONBACKEND_ModelExecute: input 'segment_ids' is GPU tensor: false
        2022-03-03 02:20:51.639491: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcublas.so.11
        I0303 02:20:52.076999 1 infer_response.cc:165] add response output: output: cls_logits, type: FP32, shape: [1,10]
        I0303 02:20:52.077032 1 http_server.cc:1071] HTTP using buffer for: 'cls_logits', size: 40, addr: 0x7f97270d2a40
        I0303 02:20:52.077040 1 tensorflow.cc:2090] TRITONBACKEND_ModelExecute: output 'cls_logits' is GPU tensor: false
        Signal (11) received.
         0# 0x0000561C823B18A9 in tritonserver
         1# 0x00007F9A844EA210 in /usr/lib/x86_64-linux-gnu/libc.so.6
         2# 0x00007F9A63602139 in /opt/tritonserver/backends/tensorflow1/libtriton_tensorflow1.so
         3# 0x00007F9A635BCBFF in /opt/tritonserver/backends/tensorflow1/libtriton_tensorflow1.so
         4# TRITONBACKEND_ModelInstanceExecute in /opt/tritonserver/backends/tensorflow1/libtriton_tensorflow1.so
         5# 0x00007F9A8507683A in /opt/tritonserver/bin/../lib/libtritonserver.so
         6# 0x00007F9A8507704D in /opt/tritonserver/bin/../lib/libtritonserver.so
         7# 0x00007F9A84F2B801 in /opt/tritonserver/bin/../lib/libtritonserver.so
         8# 0x00007F9A85070DC7 in /opt/tritonserver/bin/../lib/libtritonserver.so
         9# 0x00007F9A848D8DE4 in /usr/lib/x86_64-linux-gnu/libstdc++.so.6
        10# 0x00007F9A84D56609 in /usr/lib/x86_64-linux-gnu/libpthread.so.0
        11# clone in /usr/lib/x86_64-linux-gnu/libc.so.6
```

### **Triton Information**
**What version of Triton are you using?**
   tritonserver 21.10-py3, but I also tried 21.04, 20.08, and the above issues still occur.
**Are you using the Triton container or did you build it yourself?**
triton docker from official images

### **To Reproduce**
**Steps to reproduce the behavior.**
start triton server, and send a request.
**Describe the models (framework, inputs, outputs), ideally include the model configuration file (if using an ensemble include the model configuration file for that as well).**
1. The BERT model  we made some modify: the cls token in the last encoder of BERT is fed into a MLP and a Softmax layer to predict the 10-class probability distribution
2. model configuration file:
```
    {
        "name": "bert",
        "platform": "tensorflow_savedmodel",
        "backend": "tensorflow",
        "version_policy": {
            "latest": {
                "num_versions": 1
            }
        },
        "max_batch_size": 1,
        "input": [
            {
                "name": "input_ids",
                "data_type": "TYPE_INT32",
                "format": "FORMAT_NONE",
                "dims": [
                    128
                ],
                "is_shape_tensor": false,
                "allow_ragged_batch": false
            },
            {
                "name": "segment_ids",
                "data_type": "TYPE_INT32",
                "format": "FORMAT_NONE",
                "dims": [
                    128
                ],
                "is_shape_tensor": false,
                "allow_ragged_batch": false
            },
            {
                "name": "input_mask",
                "data_type": "TYPE_INT32",
                "format": "FORMAT_NONE",
                "dims": [
                    128
                ],
                "is_shape_tensor": false,
                "allow_ragged_batch": false
            },
            {
                "name": "label_ids",
                "data_type": "TYPE_INT32",
                "format": "FORMAT_NONE",
                "dims": [
                    128
                ],
                "is_shape_tensor": false,
                "allow_ragged_batch": false
            }
        ],
        "output": [
            {
                "name": "cls_loss",
                "data_type": "TYPE_FP32",
                "dims": [
                    10
                ],
                "label_filename": "",
                "is_shape_tensor": false
            },
            {
                "name": "cls_per_example_loss",
                "data_type": "TYPE_FP32",
                "dims": [
                    10
                ],
                "label_filename": "",
                "is_shape_tensor": false
            },
            {
                "name": "cls_logits",
                "data_type": "TYPE_FP32",
                "dims": [
                    10
                ],
                "label_filename": "",
                "is_shape_tensor": false
            },
            {
                "name": "cls_probabilities",
                "data_type": "TYPE_FP32",
                "dims": [
                    10
                ],
                "label_filename": "",
                "is_shape_tensor": false
            }
        ],
        "batch_input": [],
        "batch_output": [],
        "optimization": {
            "priority": "PRIORITY_DEFAULT",
            "input_pinned_memory": {
                "enable": true
            },
            "output_pinned_memory": {
                "enable": true
            },
            "gather_kernel_buffer_threshold": 0,
            "eager_batching": false
        },
        "instance_group": [
            {
                "name": "bert",
                "kind": "KIND_GPU",
                "count": 1,
                "gpus": [
                    0
                ],
                "secondary_devices": [],
                "profile": [],
                "passive": false,
                "host_policy": ""
            }
        ],
        "default_model_filename": "model.savedmodel",
        "cc_model_filenames": {},
        "metric_tags": {},
        "parameters": {},
        "model_warmup": []
    }
```
### **Expected behavior**
**A clear and concise description of what you expected to happen.**
Output correct results.

**Comment by tanmayv25** [comment]
@XZHUNZHUN  Can you perform a debug build of triton server and share the stack trace?
https://github.com/triton-inference-server/server/blob/main/docs/faq.md#if-the-server-segfaults-how-can-i-debug-it

Also, did you expect to use tensorflow1? Or wanted yo use TF2 with [--backend-config=tensorflow,version=2](https://github.com/triton-inference-server/tensorflow_backend#--backend-configtensorflowversionint)? 
Can you run inference successfully outside Triton i.e. directly on the model?

**Comment by Tabrizian** [comment]
Closing due to in-activity.
